### PR TITLE
Update to Laravel 11

### PR DESCRIPTION
**Pull request by eshimischi**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
         "symplify/rule-doc-generator-contracts": "^11.1",
         "nette/utils": "^3.2|^4.0",
         "sebastian/diff": "^5.1|^6.0",
-        "illuminate/container": "^10.39",
+        "illuminate/container": "^11.0",
```
Comment (spawnia) [inline composer.json]:
```suggestion
        "illuminate/container": "^10.39|^11.0",
```

Why break backwards compatibility?

Comment (eshimischi) [reply composer.json]: Yeah, it's better.

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Comment by eshimischi** [comment]
Would you please release it as new tag? https://github.com/symplify/rule-doc-generator/tags

**Comment by TomasVotruba** [comment]
This package should have downgraded release instead of locking version and releasing a new tag for every bump. 
Just noticed it's not the case. We'll need to fix that first :)

**Comment by eshimischi** [comment]
I meant this [pull353](https://github.com/BenSampo/laravel-enum/pull/353#issuecomment-1997081848) in order to receive new laravel 11 compatible package version should be upgraded or not?

**Comment by TomasVotruba** [comment]
Indeed, but we need to fix this propperly using downgrade releases as in other packages. Otherwise we might get another package change, that would force a new release, which is non-sense for tools.

File: composer.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
         "symplify/rule-doc-generator-contracts": "^11.1",
         "nette/utils": "^3.2|^4.0",
         "sebastian/diff": "^5.1|^6.0",
-        "illuminate/container": "^10.39",
+        "illuminate/container": "^10.39|^11.0",
         "webmozart/assert": "^1.11",
         "symfony/yaml": "^6.4",
         "symfony/filesystem": "^6.4"
```
Comment (owenvoke) [inline composer.json:18]: Do these need to be updated to support Symfony 7, as Laravel 11 depends uses it? 🤔

Ah, I guess it'd be resolved by https://github.com/symplify/rule-doc-generator/pull/11 👍🏻 Ignore me.

Comment (eshimischi) [reply composer.json:18]: Well i checked this package on my project which updated to Laravel 11 and both packages have version 7

**Pull request merged**